common/service-loader/src/main/java/io/helidon/common/serviceloader/HelidonServiceLoader.java

```diff
@@ -0,0 +1,360 @@
+        private boolean useSysPropExclude = true;
+        private boolean useJavaServiceLoader = true;
+        private boolean replaceImplementations = true;
+
+        private Builder(ServiceLoader<T> javaServiceLoader) {
+            this.serviceLoader = javaServiceLoader;
+        }
+
+        @Override
+        public HelidonServiceLoader<T> build() {
+            // first merge the lists together
+            List<ServiceWithPriority<T>> services = new LinkedList<>(customServices);
+            if (useJavaServiceLoader) {
+                Set<String> uniqueImplementations = new HashSet<>();
+
+                if (replaceImplementations) {
+                    customServices.stream()
+                            .map(ServiceWithPriority::instanceClassName)
+                            .forEach(uniqueImplementations::add);
+                }
+
+                serviceLoader.forEach(service -> {
+                    if (replaceImplementations) {
+                        if (!uniqueImplementations.contains(service.getClass().getName())) {
+                            services.add(new ServiceWithPriority<>(service));
+                        }
+                    } else {
+                        services.add(new ServiceWithPriority<>(service));
+                    }
+                });
+            }
+
+            if (useSysPropExclude) {
+                addSystemExcludes();
+            }
+            List<ServiceWithPriority<T>> withoutExclusions = services.stream()
+                    .filter(this::notExcluded)
+                    .collect(Collectors.toList());
+
+            // order by priority
+            return new HelidonServiceLoader<>(orderByPriority(withoutExclusions));
+        }
+
+        /**
+         * When configured to use system excludes, system property {@value #SYSTEM_PROPERTY_EXCLUDE} is used to get the
+         * comma separated list of service implementations to exclude them from the loaded list.
+         * <p>
+         * This defaults to {@code true}.
+         *
+         * @param useSysPropExclude whether to use a system property to exclude service implementations
+         * @return updated builder instance
+         */
+        public Builder<T> useSystemExcludes(boolean useSysPropExclude) {
+            this.useSysPropExclude = useSysPropExclude;
+            return this;
+        }
+
+        /**
+         * When configured to use Java Service loader, then the result is a combination of all service implementations
+         * loaded from the Java Service loader and those added by {@link #addService(Object)} or {@link #addService(Object, int)}.
+         * When set to {@code false} the Java Service loader is ignored.
+         * <p>
+         * This defaults to {@code true}.
+         *
+         * @param useJavaServiceLoader whether to use the Java Service loader
+         * @return updated builder instance
+         */
+        public Builder<T> useJavaServiceLoader(boolean useJavaServiceLoader) {
```

Comment:
This seems a little weird given that your static `builder()` and `create()` methods require a `ServiceLoader` parameter. Maybe just have variants of the static methods that do not take a `ServiceLoader`?

Reply:
That is an interesting point. The problem is that I was missing any reference to the actual type of the service. And I do not want to create a method `create(ServiceInterface.class)`, as that would mislead the user into expecting the `ServiceLoader` to be automatically used.
I only need the information when creating exceptions (as everything else is driven by generic declaration), so a string would be sufficient....

Reply:
Gotcha. So... what is the use case for disabling use of the `ServiceLoader`? Doesn't that just devolve to ordering the added services (assuming they added them with a priority) and maybe applying excludes? 

Maybe just remove this flag unless there is a good use case?

Reply:
The use case is when you want to configure only explicit services and not the ones loaded by the service loader. And yes, it devolves to ordering by priority and applying excludes.
It is an edge case, yet I wanted to support it for advanced "by hand" configurations.

Reply:
I have just realized that we already have these cases - Config.Builder supports disabling of system service loader. 
As by the time I want to disable/enable service loader I already need a builder instance of `HelidonServiceLoader`, it could be easily used in config as well. If I remove this feature, config could not utilize this library.